Project.toml

```diff
@@ -1,14 +1,17 @@
 name = "Checkpoints"
 uuid = "b4a3413d-e481-5afc-88ff-bdfbd6a50dce"
 authors = "Invenia Technical Computing Corporation"
-version = "0.3.4"
+version = "0.3.5"
```

Comment:
I'd recommend either deprecate or completely drop the old approach if this is the direction we want to go. I feel like having multiple ways of handling tags will just create more confusion.

Reply:
It is rare to set a checkpoint tag per the checkpoint .
I checked every package I had cloned locally, and none of them did it.
I am sure we must do it somewhere but I checked every use in about half a dozen packages.
Every single one passed the tags in as `tags...` where `tags` was a variable passed in.
And that is the thing that people will no longer need to do with the new way.
As such, I think it is legit to deprecate it.
There will be exceptions but having to do a `with_tags` is not that bad really

Reply:
Ok, let's deprecate the old way. I would prefer to do it in a separate PR if that's ok? And merge what we have now?